pkg/controller/bundlec/bundle_sync_task.go

```diff
@@ -303,6 +322,58 @@ func (st *bundleSyncTask) deleteRemovedResources() (retriableError bool, e error
 	return retriable, firstErr
 }
 
+func (st *bundleSyncTask) preDelete(logger *zap.Logger, name string, obj runtime.Object, m meta_v1.Object, resClient dynamic.ResourceInterface) (bool /* readyToDelete */, bool /* retriableError */, error) {
+	annotations := m.GetAnnotations()
+	deletionDelayAnnotation, ok := annotations[smith.DeletionDelayAnnotation]
+	if !ok {
+		// If there is no deletion delay annotation,
+		// we can proceed with deletion immediately
+		return true, false, nil
+	}
+
+	// Trigger the "deletion delay" logic
+	deletionDelay, err := time.ParseDuration(deletionDelayAnnotation)
+	if err != nil {
+		return false, false, errors.Wrap(err, "failed to parse deletion delay duration")
+	}
+	deletionTimestampAnnotation, ok := annotations[smith.DeletionTimestampAnnotation]
+	if !ok {
+		// Mark object with deletionTimestamp annotation to start the countdown
+		annotations[smith.DeletionTimestampAnnotation] = timeToString(time.Now())
+		m.SetAnnotations(annotations)
+
+		unstr, err := util.RuntimeToUnstructured(obj)
+		if err != nil {
+			return false, false, errors.Wrap(err, "failed to convert runtime object to unstructured")
+		}
+		_, err = resClient.Update(unstr, meta_v1.UpdateOptions{})
+		if err != nil {
+			if api_errors.IsConflict(err) {
+				// Escape the processing loop, the conflict means that
+				// we are processing a stale object revision,
+				// and a newer revision will be re-processed soon
+				return false, false, err
+			}
+			return false, true, err
+		}
+		// The object will eventually be reprocessed for the check for delay expiration
+		return false, false, nil
+	}
+	// Check if deletion delay has expired
+	deletionTimestamp, err := timeFromString(deletionTimestampAnnotation)
+	if err != nil {
+		return false, false, errors.Wrap(err, "failed to unmarshal deletion timestamp")
+	}
+	if time.Now().UTC().Before(deletionTimestamp.Add(deletionDelay)) {
```

Comment:
No need to convert to UTC. Time zone does not matter when comparing time objects because absolute timestamp values are compared.